### PR TITLE
polishing 'n cleaning code

### PR DESCRIPTION
**Pull request by ArtificialOwl**

### CODE REVIEW

File: lib/Controller/ApiController.php [hunk 1]
```diff
@@ -70,8 +72,8 @@ public function __construct(
 		$this->relatedService = $relatedService;
 		$this->configService = $configService;
 		try {
-			$this->circlesManager = \OC::$server->get(CirclesManager::class);
-		} catch (Exception $e) {
+			$this->circlesManager = OC::$server->get(CirclesManager::class);
```
Comment (blizzz) [inline lib/Controller/ApiController.php:75]:
```suggestion
			$this->circlesManager = \OCP\Server::get(CirclesManager::class);
```

use public api

File: lib/RelatedResourceProviders/CalendarRelatedResourceProvider.php [hunk 2]
```diff
@@ -64,7 +66,10 @@ public function __construct(
 		$this->urlGenerator = $urlGenerator;
 		$this->l10n = $l10n;
 		$this->calendarShareRequest = $calendarShareRequest;
-		$this->circlesManager = \OC::$server->get(CirclesManager::class);
+		try {
+			$this->circlesManager = OC::$server->get(CirclesManager::class);
```
Comment (blizzz) [inline lib/RelatedResourceProviders/CalendarRelatedResourceProvider.php:70]: also \OCP\Sever::get()

File: lib/RelatedResourceProviders/DeckRelatedResourceProvider.php [hunk 3]
```diff
@@ -67,7 +69,10 @@ public function __construct(
 		$this->urlGenerator = $urlGenerator;
 		$this->l10n = $l10n;
 		$this->deckSharesRequest = $deckSharesRequest;
-		$this->circlesManager = \OC::$server->get(CirclesManager::class);
+		try {
+			$this->circlesManager = OC::$server->get(CirclesManager::class);
```
Comment (blizzz) [inline lib/RelatedResourceProviders/DeckRelatedResourceProvider.php:73]: and here

File: lib/Controller/ApiController.php (hunk 1 above)
Comment (ArtificialOwl) [reply lib/Controller/ApiController.php:75]: wow, never heard of this, let me fix all call

Comment (blizzz) [reply lib/Controller/ApiController.php:75]: Carl added for 24 or 25.

File: lib/ILinkWeightCalculator.php [hunk 4]
```diff
@@ -33,8 +33,8 @@
 
 interface ILinkWeightCalculator {
 	/**
-	 * @param IRelatedResource[] $paths
+	 * @param IRelatedResource $current
```
Comment (Pytal) [inline lib/ILinkWeightCalculator.php:36]:
```suggestion
```
No need to duplicate this when it's already in the function declaration

File: lib/Model/DeckBoard.php [hunk 5]
```diff
@@ -106,7 +106,7 @@ public function getOwner(): string {
 	/**
 	 * @param int $lastModified
 	 *
-	 * @return DeckShare
+	 * @return DeckBoard
```
Comment (Pytal) [inline lib/Model/DeckBoard.php:109]:
```suggestion
```
Already in the function declaration as well

File: lib/RelatedResourceProviders/TalkRelatedResourceProvider.php [hunk 6]
```diff
@@ -183,7 +192,7 @@ private function processRoomParticipant(RelatedResource $related, TalkActor $act
 
 
 	/**
-	 * @param TalkRoom $actor
+	 * @param TalkActor $actor
```
Comment (Pytal) [inline lib/RelatedResourceProviders/TalkRelatedResourceProvider.php:195]:
```suggestion
```
Also in the function declaration already so not needed

File: lib/Service/RelatedService.php [hunk 7]
```diff
@@ -451,14 +451,14 @@ private function filterUnavailableResults(array $result): array {
 	}
 
 	/**
-	 * @param IRelatedResource[] $paths
+	 * @param IRelatedResource $current
```
Comment (Pytal) [inline lib/Service/RelatedService.php:454]:
```suggestion
```
Same as above

File: lib/Model/DeckBoard.php (hunk 5 above)
Comment (ArtificialOwl) [reply lib/Model/DeckBoard.php:109]: we already had a discussion about this, we keep phpdoc

File: lib/ILinkWeightCalculator.php (hunk 4 above)
Comment (ArtificialOwl) [reply lib/ILinkWeightCalculator.php:36]: we already had a discussion about this, we keep phpdoc

File: lib/RelatedResourceProviders/TalkRelatedResourceProvider.php (hunk 6 above)
Comment (ArtificialOwl) [reply lib/RelatedResourceProviders/TalkRelatedResourceProvider.php:195]: we already had a discussion about this, we keep phpdoc

File: lib/Service/RelatedService.php (hunk 7 above)
Comment (ArtificialOwl) [reply lib/Service/RelatedService.php:454]: we already had a discussion about this, we keep phpdoc

File: lib/LinkWeightCalculators/AncienShareWeightCalculator.php [hunk 8]
```diff
@@ -46,46 +47,38 @@ class AncienShareWeightCalculator implements ILinkWeightCalculator {
...
-				// calculate an improvement base on 0.75 up to 1.2, based on difference of time between 2 shares
-				// with 1.0 score for a 3 month period
-				$neutral = 90 * 24 * 3600;
-				$ratio = $diff - $neutral;
-				$impr = 1 - ($ratio * 0.2 / $neutral);
-				$impr = max($impr, 0.75);
-				$entry->improve($impr, 'ancien_3m');
+			$now = time();
+			$entryCreation = $entry->getMetaInt(RelatedResource::LINK_CREATION);
+			if ($entryCreation < $now - (5 * 360 * 24 * 3600)) { // 3y
```
Comment (come-nc) [inline lib/LinkWeightCalculators/AncienShareWeightCalculator.php:62]:
```suggestion
			if ($entryCreation < $now - (5 * 360 * 24 * 3600)) { // 5y
```

File: lib/LinkWeightCalculators/AncienShareWeightCalculator.php [hunk 9]
```diff
@@ -46,46 +47,38 @@ class AncienShareWeightCalculator implements ILinkWeightCalculator {
...
+				$entry->improve(self::$RATIO_5Y, 'ancien_5y');
+			} elseif ($entryCreation < $now - (3 * 360 * 24 * 3600)) { // 3y
+				$entry->improve(self::$RATIO_3Y, 'ancien_3y');
+			} elseif ($entryCreation < $now - (360 * 24 * 3600)) { // 1y
+				$entry->improve(self::$RATIO_3Y, 'ancien_1y');
```
Comment (come-nc) [inline lib/LinkWeightCalculators/AncienShareWeightCalculator.php:67]: Why is ratio 3 years here?

**Review by come-nc — APPROVED** [review]

**Pull request merged**